packages/babel-plugin-transform-typescript/src/index.ts

```diff
@@ -371,11 +371,13 @@ export default declare((api, opts: Options) => {
             }
 
             if (!onlyRemoveTypeImports && stmt.isTSImportEqualsDeclaration()) {
-              const { id, isExport } = stmt.node;
+              const { id } = stmt.node;
               const binding = stmt.scope.getBinding(id.name);
               if (
                 binding &&
-                !isExport &&
+                (process.env.BABEL_8_BREAKING ||
+                  // @ts-ignore(Babel 7 vs Babel 8) Babel 7 AST
+                  !stmt.node.isExport) &&
```

Comment:
Here we are checking top level statements, since a top-level TSImportEqualsDeclaration will be always an import declaration in Babel 8, we don't need the `!isExport` check here.